test/test_helpers/events.rb

```diff
@@ -13,6 +13,15 @@ def check_events_validity!(type:, count: 1, opts: {})
 
       assert_equal count.to_i, specific_events.count
     end
+
+    # Since Rails 6.1 the EventStore::Event records (among others) are added to the
+    # list of transaction objects, and are rolled back on the transaction rollback.
+    # The problem is that this triggers the Event's `load` and `deserialize`, and if
+    # the object itself is rolled back first, the deserialization of the related event
+    # will fail. This causes some tests (that have explicit transactions and rollbacks) to fail.
+    def stub_event_rolledback!
+      EventStore::Event.any_instance.stubs(:rolledback!)
+    end
```

Comment:
I think I understand the change that introduced this now https://github.com/rails/rails/commit/77f7b2df3aa3b7eb318cc830f239fd5ec2a7f28f#diff-8dd03b7fb9b72a3bd338955c1de75652d60453230c6544f3851c0d0b3746a675L345-R349

Previously the AR objects lacking has_transactional_callbacks? were not added to the transaction, while now all are added.

This fix is alright during testing. But what if a transaction fails in production? Wouldn't `EventStore::Event` objects still fail to rollback and probably messing up the whole transaction rollback and reporting an unhelpful error message in bugsnag?